GPU/Common/TextureCacheCommon.cpp

```diff
@@ -154,46 +154,48 @@ void TextureCacheCommon::GetSamplingParams(int &minFilt, int &magFilt, bool &sCl
 
 	GETexLevelMode mipMode = gstate.getTexLevelMode();
 	mode = mipMode;
+
 	bool autoMip = mipMode == GE_TEXLEVEL_MODE_AUTO;
+
+	// TODO: Slope mipmap bias is still not well understood.
 	lodBias = (float)gstate.getTexLevelOffset16() * (1.0f / 16.0f);
 	if (mipMode == GE_TEXLEVEL_MODE_SLOPE) {
 		lodBias += 1.0f + TexLog2(gstate.getTextureLodSlope()) * (1.0f / 256.0f);
 	}
 
 	// If mip level is forced to zero, disable mipmapping.
 	bool noMip = maxLevel == 0 || (!autoMip && lodBias <= 0.0f);
-	if (IsFakeMipmapChange())
+	if (IsFakeMipmapChange()) {
 		noMip = noMip || !autoMip;
+	}
 
 	if (noMip) {
 		// Enforce no mip filtering, for safety.
 		minFilt &= 1; // no mipmaps yet
 		lodBias = 0.0f;
 	}
 
-	if (g_Config.iTexFiltering == TEX_FILTER_LINEAR_VIDEO) {
-		bool isVideo = videos_.find(addr & 0x3FFFFFFF) != videos_.end();
-		if (isVideo) {
+	if (!(magFilt & 1) && addr != 0 && g_Config.iTexFiltering == TEX_FILTER_LINEAR_VIDEO) {
+		if (videos_.find(addr & 0x3FFFFFFF) != videos_.end()) {
 			magFilt |= 1;
 			minFilt |= 1;
 		}
 	}
-	if (g_Config.iTexFiltering == TEX_FILTER_LINEAR && (!gstate.isColorTestEnabled() || IsColorTestTriviallyTrue())) {
-		if (!gstate.isAlphaTestEnabled() || IsAlphaTestTriviallyTrue()) {
+
+	// Filtering overrides
+	if (g_Config.iTexFiltering == TEX_FILTER_LINEAR) {
+		// Only override to linear filtering if there's no alpha or color testing going on.
+		if ((!gstate.isColorTestEnabled() || IsColorTestTriviallyTrue()) &&
+		    (!gstate.isAlphaTestEnabled() || IsAlphaTestTriviallyTrue())) {
 			magFilt |= 1;
 			minFilt |= 1;
 		}
-	}
-	bool forceNearest = g_Config.iTexFiltering == TEX_FILTER_NEAREST;
-	// Force Nearest when color test enabled and rendering resolution greater than 480x272
-	if ((gstate.isColorTestEnabled() && !IsColorTestTriviallyTrue()) && g_Config.iInternalResolution != 1 && gstate.isModeThrough()) {
+	} else if (g_Config.iTexFiltering == TEX_FILTER_NEAREST ||
+		(gstate.isModeThrough() && g_Config.iInternalResolution != 1 &&
+		gstate.isColorTestEnabled() && !IsColorTestTriviallyTrue() && gstate.getColorTestRef() != 0)) {
```

Comment:
My opinion is that multiline ifs, and multiline statements in general, is just a sign of hard to read code.

I don't think anyone will look at this if statement and suggest that its logic is obvious or clear.  Adding more newlines, parenthesis, or playing with indentation would not help that at all.  If you can't get your point across in a normal sentence, making it longer without a period isn't going to improve the situation.

Generally, I think the same about function arguments (calls or defs), etc.  If your function takes 10 arguments, you should use a struct/class, if your if has 10 clauses you should use temporaries, funcs, or lambdas, etc.

-[Unknown]

Reply:
Yeah I'm honestly a bit torn here. Sequences of only && are pretty easy to read chained but as soon as there's an ||, you start to need the parenthesis and it suddenly becomes a lot less readable.

I'm going to change it to a kind of middle ground where each if has only one type of operator. That should take care of it.